### ISSUE TITLE
Some amounts overflow in XDR

### ISSUE THREAD

**Issue by umdemir** [Soneso/stellar-ios-mac-sdk#27]
Some amounts overflow when they are encoded in XDR.

**Sample amount:** 
```
▿ 0.18050270000000002048
  ▿ _mantissa : 8 elements
    - .0 : 59392
    - .1 : 14819
    - .2 : 28910
    - .3 : 64127
    - .4 : 0
    - .5 : 0
    - .6 : 0
    - .7 : 0
```

**Encoded Amount:**  -0.0039647 (raw: -39647)

**Sample Response:** 
```
{\n  \"type\": \"https://stellar.org/horizon-errors/transaction_failed\",\n  \"title\": \"Transaction Failed\",\n  \"status\": 400,\n  \"detail\": \"The transaction failed when submitted to the stellar network. The `extras.result_codes` field on this response contains further details.  Descriptions of each code can be found at: https://www.stellar.org/developers/learn/concepts/list-of-operations.html\",\n  \"extras\": {\n    \"envelope_xdr\": \"AAAAAJoY8kcE4G/6PqnzgNutcZ6Ffu22hOTbo1OS0RRhSLEZAAAAZACfbX4AAABgAAAAAAAAAAAAAAABAAAAAQAAAACaGPJHBOBv+j6p84DbrXGehX7ttoTk26NTktEUYUixGQAAAAEAAAAAdxM91EE5PqEiQmQg9oR2q2gxA50UA47qDj8GlYjrenQAAAABT0cAAAAAAAB8CTr1M3808ueF/pik9o3XLuo85lEv/PQVDVgGdMxkxP///////2UhAAAAAAAAAAFhSLEZAAAAQFSjH9sG6N0VgHu7KcPc+ICOCpy9eWwgpBhZMyhVwKVdQYSJVYuEL2H+uGJVf76rJa1losFNYYlFW6QJl5ImLQ0=\",\n    \"result_codes\": {\n      \"transaction\": \"tx_failed\",\n      \"operations\": [\n        \"op_malformed\"\n      ]\n    },\n    \"result_xdr\": \"AAAAAAAAAGT/////AAAAAQAAAAAAAAAB/////wAAAAA=\"\n  }\n}
```

**Comment by umdemir** [comment]
https://github.com/Soneso/stellar-ios-mac-sdk/pull/26

**Comment by chelemen-razvan** [comment]
Fix in 1.4.2
